src/magneticField/CMZField.cpp

```diff
@@ -0,0 +1,441 @@
+double CMZField::BxAz(double x, double y, double z, double m, double B1, double eta, double R, double rr) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    return (cos(phi)-1./eta * sin(phi)) *BrAz(r,phi,z,m,B1,eta,R,rr);
+}
+
+double CMZField::ByAz(double x, double y, double z, double m, double B1, double eta, double R, double rr) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    return (sin(phi)+1./eta*cos(phi))*BrAz(r,phi,z,m,B1,eta,R,rr);
+}
+
+// Polodial field in cartesian coordinates// for non-thermal filament and the intercloud region
+double CMZField::ByPol(double x, double y, double z, double B1, double a1, double a2, double eta) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    double B2=B1/eta;// B1 is the observed magnetic field strength and eta the normalization factor otained from the integrations due to the normalization
+    double L=scale(a2/2.,1);
+    double a= 1/pow(scale(a1/2.,2),2.);
+    return sin(phi)*Br(r,z,B2,a,L);
+}
+
+double CMZField::BxPol(double x, double y, double z, double B1, double a1, double a2, double eta) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    double B2=B1/eta;
+    double L=scale(a2/2.,1);
+    double a= 1/pow(scale(a1/2.,2),2.);
+    return cos(phi)*Br(r,z,B2,a,L);
+}
+
+double CMZField::BzPol(double x, double y, double z, double B1, double a1, double a2, double eta) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    double B2=B1/eta;
+    double L=scale(a2/2.,1);
+    double a= 1/pow(scale(a1/2.,2),2.);
+    return Bz(r,z,B2,a,L);
+}
+/////end model C + azimuthal component
+
+bool CMZField::getUseMCField() const {
+    return useMCField;
+}
+bool CMZField::getUseICField() const {
+    return useICField;
+}
+bool CMZField::getUseNTFField() const {
+    return useNTFField;
+}
+bool CMZField::getUseRadioArc() const {
+    return useRadioArc;
+}
+
+void CMZField::setUseMCField(bool use) {
+    useMCField = use;
+}
+void CMZField::setUseICField(bool use) {
+    useICField = use;
+}
+void CMZField::setUseNTFField(bool use) {
+    useNTFField = use;
+}
+void CMZField::setUseRadioArc(bool use) {
+    useRadioArc = use;
+}
+
+// Parameter identification:
+// r: radius in cylindrical coordinates
+// z= z component in cylindircal coordintes
+// phi: phi component in cylindircal coordinates
+// B1: observed magnetic field strength
+// B2: normalization magnetic field strength according to B1
+// m: azimuthal wavenumber m=0 for axissymmetric, m=1 for bisymmetric and m=2 quadrosymmetric, ... field lines
+// a: stricly positive free parameters governing the opening of field lines away from the z-axis
+// L: exponential scale length length of the cloud radius Longitudinal extent
+// Lp:scale length of the winding function set to a fraction of the cloud radius  Longitudes of peaks mainly at high |l|
```

Comment:
"Lp: scale length of the winding function set to a fraction of the cloud radius longitudes of peaks mainly at high |l|"
I do not get this sentence. Please reword or clarify.